test/memoryPoolAPI.cpp

```diff
@@ -0,0 +1,307 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// This file contains tests for UMF pool API
+
+#include "pool.h"
+#include "pool.hpp"
+#include "provider.h"
+#include "provider.hpp"
+
+#include "memoryPool.hpp"
+#include "umf/memory_provider.h"
+
+#include <array>
+#include <string>
+#include <thread>
+#include <unordered_map>
+
+using umf_test::test;
+
+TEST_F(test, memoryPoolTrace) {
+    static std::unordered_map<std::string, size_t> poolCalls;
+    static std::unordered_map<std::string, size_t> providerCalls;
+    auto tracePool = [](const char *name) { poolCalls[name]++; };
+    auto traceProvider = [](const char *name) { providerCalls[name]++; };
+
+    auto nullProvider = umf_test::wrapProviderUnique(nullProviderCreate());
```

Comment:
I'm not convinced about using wrapProviderUnique function instead of using regular UMF API (umfProviderCreate, umfPoolCreate) here. I think it complicates the code and we also don't test here umfProviderCreate and umfPoolCreate functions straightly.

Reply:
wrapProvider unique does not call any create function underneath, it only wraps the handle into unique_ptr - you still need to call umfProviderCreate (or nullProviderCreate as in this case) so we don't hide anything